pkg/cloudclient/aws/private.go

```diff
@@ -62,35 +62,60 @@ var (
 	userdataEndVerifier   string = "USERDATA END"
 )
 
-func newClient(ctx context.Context, logger ocmlog.Logger, accessID, accessSecret, sessiontoken, region, instanceType string, tags map[string]string) (*Client, error) {
-	cfg, err := config.LoadDefaultConfig(ctx,
-		config.WithRegion(region),
-		config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
-			Value: aws.Credentials{
-				AccessKeyID: accessID, SecretAccessKey: accessSecret, SessionToken: sessiontoken,
-			},
-		}),
-	)
+func getEc2ClientFromInput(input ClientInput) (ec2.Client, error) {
```

Comment:
At first glance, `aws.go/ GetEc2ClientFromInput()` and `private.go/ getEc2ClientFromInput()` has pretty much same logic.
It looks a bit repeating itself so it would be nice if we unified it